appscale/tools/appscale.py

```diff
@@ -763,12 +758,20 @@ def destroy(self):
     if "EC2_URL" in contents_as_yaml:
       os.environ["EC2_URL"] = contents_as_yaml["EC2_URL"]
 
-    if 'keyname' in contents_as_yaml:
-      command.append("--keyname")
-      command.append(contents_as_yaml['keyname'])
+    if clean:
+      if 'test' not in contents_as_yaml or contents_as_yaml['test'] != True:
+        LocalState.confirm_or_abort("Clean will delete every data in the deployment.")
+      all_ips = LocalState.get_all_public_ips(keyname)
+      for ip in all_ips:
```

Comment:
@obino I forgot that we wanted to make the next step removing the public ips. making it threaded right now isn't necessary.